toqito/matrix_ops/commutant.py

```diff
@@ -0,0 +1,60 @@
+"""Module for computing the commutant of a set of matrices."""
+
+import numpy as np
+from scipy.linalg import null_space
+
+
+def commutant(A: np.ndarray | list[np.ndarray]) -> list[np.ndarray]:
+    r"""Compute an orthonormal basis for the commutant algebra.
+
+    Given a set of matrices A, this function determines an orthonormal basis 
+    (with respect to the Hilbert-Schmidt inner product) for the algebra of matrices 
+    that commute with every matrix in A.
+
+
+    The commutant condition is given by:
+
+    .. math::
+        A X = X A \quad \forall A \in \mathcal{A}.
+
+    This can be rewritten as:
+
+    .. math::
+        (A \otimes I - I \otimes A^T) \text{vec}(X) = 0.
+
+    where :math:`\text{vec}(X)` denotes the vectorization of :math:`X`. The null space of this
+    equation provides a basis for the commutant.
+
+    References
+    ==========
+    .. bibliography::
+        :filter: docname in docnames
```

Comment:
If you are not citing a reference, you don't really need this section.

Reply:
Yes, but I believe there should be a reference to something (even just a Wikipedia page, the QETLAB reference, etc. that would be relevant here).